oneview/resource_logical_interconnect_group.go

```diff
@@ -1796,10 +2125,45 @@ func resourceLogicalInterconnectGroupRead(d *schema.ResourceData, meta interface
 				}
 				uplinkSets[i]["logical_port_config"] = logicalPortConfigInfo
 			}
-			d.Set("uplink_set", uplinkSets)
+
+		}
+	}
+
+	//Oneview send the uplink set in unordered way so ordering it.
+	matchedUplinkSets := make([]map[string]interface{}, 0)
+	unmatchedUplinkSets := make([]map[string]interface{}, 0)
+
+	if val, ok := d.GetOk("uplink_set"); ok {
+		oneviewuplinksetCount := len(uplinkSets)
+
+		uss := val.([]interface{})
+		uplinksetCount := len(uss)
+		for i := 0; i < uplinksetCount; i++ {
+			currName := (uss[i].(map[string]interface{}))["name"]
+			for j := 0; j < oneviewuplinksetCount; j++ {
+
+				if currName == uplinkSets[j]["name"] {
```

Comment:
i think its better
if uplinkSets[j]!= Nil and currName == uplinkSets[j]["name"]

Also, is OV uplinkset case insensitive? in that case better to make both of them to same case before comparison

Reply:
Done